lib/linter/linter.js

```diff
@@ -1408,12 +1394,14 @@ class Linter {
             ? getDirectiveComments(sourceCode, ruleId => getRule(slots, ruleId), options.warnInlineConfig, config)
             : { configuredRules: {}, enabledGlobals: {}, exportedVariables: {}, problems: [], disableDirectives: [] };
 
-        // augment global scope with declared global variables
-        addDeclaredGlobals(
-            sourceCode.scopeManager.scopes[0],
-            configuredGlobals,
-            { exportedVariables: commentDirectives.exportedVariables, enabledGlobals: commentDirectives.enabledGlobals }
-        );
+        // augment global scope with declared global variables for ESTree only
+        if (sourceCode.isESTree) {
+            addDeclaredGlobals(
+                sourceCode.scopeManager.scopes[0],
+                configuredGlobals,
+                { exportedVariables: commentDirectives.exportedVariables, enabledGlobals: commentDirectives.enabledGlobals }
+            );
+        }
```

Comment:
```suggestion
        addDeclaredGlobals(
            sourceCode.scopeManager.scopes[0],
            configuredGlobals,
            { exportedVariables: commentDirectives.exportedVariables, enabledGlobals: commentDirectives.enabledGlobals }
            );
```

This one too.

Reply:
Ah, missed that. Can't wait til we can delete the duplicate logic.